processor/src/operations/horner_ops.rs

```diff
@@ -0,0 +1,515 @@
+use vm_core::{Felt, Operation};
+
+use crate::{ExecutionError, Process, QuadFelt};
+
+// HORNER EVALUATION OPERATIONS
+// ================================================================================================
+
+impl Process {
+    // HORNER EVALUATION WITH COEFFICIENTS OVER BASE FIELD
+    // --------------------------------------------------------------------------------------------
+
+    /// Performs 8 steps of the Horner evaluation method on a polynomial with coefficients over
+    /// the base field, i.e., it computes
+    ///
+    /// acc' = (((acc_tmp * alpha + c3) * alpha + c2) * alpha + c1) * alpha + c0
+    ///
+    /// where
+    ///
+    /// acc_tmp := (((acc * alpha + c7) * alpha + c6) * alpha + c5) * alpha + c4
+    ///
+    ///
+    /// In other words, the intsruction computes the evaluation at alpha of the polynomial
+    ///
+    /// P(X) := c7 * X^7 + c6 * X^6 + ... + c1 * X + c0
+    ///
+    /// As can be seen from the two equations defining acc', the instruction can be used in order
+    /// to compute the evaluation of polynomials of arbitrary degree by repeated invocations of
+    /// the same instruction interleaved with any operation that loads the next batch of 8
+    /// coefficients on the top of the operand stack, i.e., `mem_stream` or `adv_pipe`.
+    ///
+    /// The stack transition of the instruction can be visualized as follows:
+    ///
+    /// Input:
+    ///
+    /// +------+------+------+------+------+------+------+------+---+---+---+---+------+------+------+------+
+    /// |  c0  |  c1  |  c2  |  c3  |  c4  |  c5  |  c6  |  c7  | - | - | - | - |c_addr|r_addr| acc1 | acc0 |
```

Comment:
nit: `c_addr` is actually not used by this instruction, so in isolation, could be anything and has no meaning. So I think it'd be better not to mention it here?

Reply:
Good point, removed